app/models/user.rb

```diff
@@ -9,17 +12,31 @@ class User < ActiveRecord::Base
 
   enum role: {admin: 0, member: 1}
 
-  # In order to override the devise validations, I have to remove the validatable module
-  # and re-implement it with some changes
+  # In order to override the devise validations to scope by deleted_at, we have to remove
+  # the validatable module and re-implement it with our changes.
   validates :email, presence: true
   validates :email, format: {with: Devise.email_regexp, allow_blank: true}, if: :email_changed?
   validates :email, uniqueness: {scope: :deleted_at}, unless: :deleted?
   validates :password, presence: true, confirmation: true, on: :create
   validates :password, length: {within: Devise.password_length, allow_blank: true}
 
   # Non-devise validations
+  validates :given_names, :family_name, length: {maximum: MAX_NAME_LENGTH}, presence: true
```

Comment:
Since we are now sanitizing names with devise, wouldn't you consider them to be devise validations?

Further, I wouldn't enforce a family name since some people/cultures don't have them.

From the article:
>Similarly, don't require that people supply a family name. In cultures such as parts of Southern India, Malaysia and Indonesia, a large number of people have names that consist of a given name only, with no patronym. If you require family names, you may create significant problems in these cultures, as users enter garbage data like "." or "Mr." in the family name field just to escape the form.